### PR TITLE
MM-60974 - Message Export: Add performance metrics

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Add performance metrics to the message export system. Only used in the CLI for now, but we can expose these as needed in the UI. 

- Linked PR: https://github.com/mattermost/mattermost-compliance-export/pull/19

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-60974

#### Release Note
This is going into the Message export feature branch, no release note needed.

```release-note
NONE
```

### CODE REVIEW

**Review by BenCookie95 — APPROVED** [review]

File: server/enterprise/message_export/actiance_export/actiance_export.go [hunk 1]
```diff
@@ -364,18 +369,19 @@ func buildChannelExport(startTime int64, endTime int64, channel *shared.Metadata
 	return &channelExport
 }
 
-func writeExport(rctx request.CTX, export *RootNode, uploadedFiles []*model.FileInfo, exportBackend filestore.FileBackend, fileAttachmentBackend filestore.FileBackend, batchPath string) (int, error) {
+func writeExport(rctx request.CTX, export *RootNode, uploadedFiles []*model.FileInfo, exportBackend filestore.FileBackend, fileAttachmentBackend filestore.FileBackend, batchPath string) (res shared.WriteExportResult, err error) {
```
Comment (hanzei) [inline server/enterprise/message_export/actiance_export/actiance_export.go:372]: Do we need the named return values?

File: server/enterprise/message_export/shared/shared.go [hunk 2]
```diff
@@ -59,13 +65,21 @@ type BackendParams struct {
 	HtmlTemplates         *templates.Container
 }
 
+type WriteExportResult struct {
+	TransferringFilesMs int64
+	ProcessingXmlMs     int64
+	TransferringZipMs   int64
```
Comment (hanzei) [inline server/enterprise/message_export/shared/shared.go:71]: 0/5 the `Ms` suffix is a bit hard to understand. Maybe add a comment and remove the `Ms` part?

**Review by hanzei — APPROVED** [review]
Nice work. A few non blocking questions.

File: server/enterprise/message_export/actiance_export/actiance_export.go (hunk 1 above)
Comment (cpoile) [reply server/enterprise/message_export/actiance_export/actiance_export.go:372]: I guess not, removed, thanks.

File: server/enterprise/message_export/shared/shared.go (hunk 2 above)
Comment (cpoile) [reply server/enterprise/message_export/shared/shared.go:71]: I think that was a comment from an earlier PR, so I think I'll keep them for clarity about the units.

**Comment by cpoile** [comment]
Merging because it's the same flaky test over and over that is hopefully fixed on master.

**Pull request merged**